csharp/autobuilder/Semmle.Autobuild.Shared/MsBuildRule.cs

```diff
@@ -10,15 +10,13 @@ internal static class MsBuildCommandExtensions
         /// <summary>
         /// Appends a call to msbuild.
         /// </summary>
-        /// <param name="cmdBuilder"></param>
-        /// <param name="builder"></param>
         /// <returns></returns>
-        public static CommandBuilder MsBuildCommand(this CommandBuilder cmdBuilder, IAutobuilder<AutobuildOptionsShared> builder)
+        public static CommandBuilder MsBuildCommand(this CommandBuilder cmdBuilder, IAutobuilder<AutobuildOptionsShared> builder, bool preferDotnet)
         {
             // mono doesn't ship with `msbuild` on Arm-based Macs, but we can fall back to
             // msbuild that ships with `dotnet` which can be invoked with `dotnet msbuild`
             // perhaps we should do this on all platforms?
-            return builder.Actions.IsRunningOnAppleSilicon()
+            return builder.Actions.IsRunningOnAppleSilicon() || preferDotnet
```

Comment:
Nit/Scope creep: There is a possible performance optimization, where we can avoid making multiple calls to `sysctl machdep.cpu.brand_string`.
In `IBuildActions` we could declare a method that returns the value of `builder.Actions.IsRunningOnAppleSilicon() || !builder.Actions.IsWindows() && !builder.Actions.IsMonoInstalled();` and stores it in backing field for subsequent uses. (It is the same condition that is used on l. 19 and l. 85).
Also perfectly fine, if you don't want to change it 😄